### PR TITLE
Fetch clinic detail when using v2 endpoints

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This PR
- Adds a fetch to the vaos service clinics endpoint
- Sets clinic name from the fetched clinic data, since we don't have that on appointments from vaos service
- Adds tests for both scenarios, where we successfully and unsuccessfully fetch clinic details

## Original issue(s)
department-of-veterans-affairs/va.gov-team#27642

## Testing done
Unit and local testing

## Acceptance criteria
- [ ] Clinic is fetched

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/services/patient/index.js [hunk 1]
```diff
@@ -151,6 +151,7 @@ async function fetchPatientEligibilityFromVAR({
  * @param {'direct'|'request'|null} [params.type=null] The type to check eligibility for. By default,
  *   will check both
  * }
+ * @param {boolean} [params.useV2=false] Use the v2 apis when making eligibility calls
```
Comment (jbalboni) [inline src/applications/vaos/services/patient/index.js:154]: I forgot to fix these before merging my other PR this morning

File: src/applications/vaos/services/mocks/index.js [hunk 2]
```diff
@@ -265,8 +265,14 @@ const responses = {
     });
   },
   'GET /vaos/v2/locations/:id/clinics': (req, res) => {
-    // Will need to modify this logic when we fetch specific clinics
-    // for the detail page
+    if (req.params.clinic_ids) {
```
Comment (LeKeve) [inline src/applications/vaos/services/mocks/index.js:268]: These should be `req.query.clinic_ids`.

File: src/applications/vaos/services/healthcare-service/index.js [hunk 3]
```diff
@@ -82,6 +82,30 @@ export async function getSupportedHealthcareServicesAndLocations({
...
+ * @param {string} params.id The clinic id
+ */
+export async function fetchHealthcareServiceById({ locationId, id }) {
+  try {
+    const results = await getClinics({
+      locationId,
+      clinicIds: [id],
+    });
+
+    return transformClinicsV2(results)[0];
```
Comment (LeKeve) [inline src/applications/vaos/services/healthcare-service/index.js:100]: Is the `[0]` necessary? The v2 endpoint should only return a single clinic if `clinicIds` is passed a single id array.

**Review by LeKeve — CHANGES_REQUESTED** [review]

File: src/applications/vaos/services/healthcare-service/index.js (hunk 3 above)
Comment (LeKeve) [reply src/applications/vaos/services/healthcare-service/index.js:100]: Disregard this. I misread as `results[0]` and thought we were passing the first clinic we got in the response. This is actually grabbing the only clinic in the transformed array. Looks good.

**Review by LeKeve — APPROVED** [review]

**Pull request merged**